components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -772,7 +775,14 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
         let apply_time = APPLY_TIME_HISTOGRAM.local();
         for (ch, resp) in callbacks.drain(..) {
             for tracker in ch.write_trackers() {
-                tracker.observe(now, &apply_time, |t| &mut t.metrics.apply_time_nanos);
+                let mut apply_wait_nanos = 0_u64;
```

Comment:
It's already recorded in `t.metrics.apply_time_nanos`.

Reply:
Yes, today the apply_time_nanos includes the apply wait time and apply log duration. But there's STORE_APPLY_LOG_HISTOGRAM for apply log duration. We need to fill this metrics as well to make it consistent with v1.

Reply:
> It's already recorded in `t.metrics.apply_time_nanos`.

Yes, today the apply_time_nanos includes the apply wait time and apply log duration. But there's STORE_APPLY_LOG_HISTOGRAM for apply log duration. We need to fill this metrics as well to make it consistent with v1.

Reply:
Now that waterfall metrics is enabled by default, we can just delete those graph or use the same expression from waterfall metrics.

Reply:
> Now that waterfall metrics is enabled by default, we can just delete those graph or use the same expression from waterfall metrics.

But educating all of our customers will be hard and to avoid unnecessary support cost, I think we should keep the compatibility.

Reply:
Customers may probably not look into the metrics. Using the same expression from waterfall metrics can also keep the compatibility without introducing extra cost. Either way is OK to me.